src/gui/guiFormSpecMenu.cpp

```diff
@@ -3453,22 +3508,21 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 {
 	if (event.EventType==EET_KEY_INPUT_EVENT) {
 		KeyPress kp(event.KeyInput);
-		if (event.KeyInput.PressedDown && (
-				(kp == EscapeKey) || (kp == CancelKey) ||
-				((m_client != NULL) && (kp == getKeySetting("keymap_inventory"))))) {
-			tryClose();
-			return true;
-		}
 
-		if (m_client != NULL && event.KeyInput.PressedDown &&
-				(kp == getKeySetting("keymap_screenshot"))) {
-			m_client->makeScreenshot();
-		}
-		if (event.KeyInput.PressedDown &&
-			(event.KeyInput.Key==KEY_RETURN ||
-			 event.KeyInput.Key==KEY_UP ||
-			 event.KeyInput.Key==KEY_DOWN)
-			) {
+		if (event.KeyInput.PressedDown) {
+			current_keys_pending.pending_key_event = true;
+			current_keys_pending.key_event = kp;
+
+			if ((kp == EscapeKey) || (kp == CancelKey) ||
+					((m_client != NULL) && (kp == getKeySetting("keymap_inventory")))) {
+				tryClose();
+				return true;
+			}
+
+			if (m_client != NULL && (kp == getKeySetting("keymap_screenshot"))) {
```

Comment:
```suggestion
			if (m_client && kp == getKeySetting("keymap_screenshot")) {
```

Reply:
On a side-note: Does this fix the problem of `keymap_screenshot` keypress being ignored when a formspec is open?

Reply:
Is it ignored?  I can take screenshots while formspecs are open, and it is returned properly in `key_mapped`, so I don't see the problem.

Reply:
It's ignored in the main menu. Or maybe I'm wrong about this. It's probably initialized only when a game is launched. I assumed it had something to do with formspecs, sorry. :)